huaweicloud/services/sfsturbo/resource_huaweicloud_sfs_turbo_change_charge_mode.go

```diff
@@ -0,0 +1,183 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+			"share_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"period": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"period_unit": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"auto_renew": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"is_auto_pay": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enable_force_new": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"true", "false"}, false),
+				Description:  utils.SchemaDesc("", utils.SchemaDescInput{Internal: true}),
+			},
+		},
+	}
+}
+
+func buildSFSTurboChangeChargeModeBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"bss_param": map[string]interface{}{
+			"period_num":    d.Get("period"),
+			"period_type":   d.Get("period_unit"),
+			"is_auto_renew": changeStringToInt(utils.ValueIgnoreEmpty(d.Get("auto_renew")).(string)),
+			"is_auto_pay":   changeStringToInt(utils.ValueIgnoreEmpty(d.Get("is_auto_pay")).(string)),
+		},
+	}
+
+	return bodyParams
+}
+
+func changeStringToInt(str string) interface{} {
+	if str == "" {
+		return nil
+	}
+	resp, err := strconv.Atoi(str)
+	if err != nil {
+		return nil
+	}
+
+	return resp
+}
+
+func resourceSFSTurboChangeChargeModeCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		shareId = d.Get("share_id").(string)
+		httpUrl = "v2/{project_id}/sfs-turbo/shares/{share_id}/change-charge-mode"
+		product = "sfs-turbo"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating SFS Turbo client: %s", err)
+	}
+
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{share_id}", shareId)
+
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildSFSTurboChangeChargeModeBodyParams(d)),
+	}
+
+	resp, err := client.Request("POST", requestPath, &requestOpt)
+	if err != nil {
+		return diag.Errorf("error changing SFS Turbo charge mode: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	orderIds := utils.PathSearch("order_ids", respBody, []interface{}{})
+	if len(orderIds.([]interface{})) == 0 {
+		return diag.Errorf("error getting order IDs from response: %v", respBody)
+	}
+
+	bssClient, err := cfg.BssV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating BSS v2 client: %s", err)
+	}
+
+	firstOrderId := orderIds.([]interface{})[0].(string)
```

Comment:
From a rigorous perspective, it is necessary to determine whether firstOrderId has a value.